### PR TITLE
BUG: Fix difficult to see slice offset text in Slicer dark mode

### PR DESCRIPTION
**Pull request by jamesobutler**
This is work for #7233.

https://github.com/Slicer/Slicer/commit/8e70f763b0b351db00d8851871e68f11469d1a2e introduced this issue of light text being used for the slice offset text when in Slicer dark mode. Dark text was correctly being used in Slicer light mode. This change uses dark text regardless of current Slicer style.

|`main` | This PR |
|--------|---------|
|image|image|

### CODE REVIEW

**Review by lassoan — COMMENTED** [review]
Thank you for the fix.

Why this hardcoding to black was not necessary before? Had the color been set using palette before? If stylesheet overwrites palette colors anyway then could we keep the palette-based coloring? We should not patch coloring issues using stylesheet when it can be fixed using the currently official coloring method (by using palettes). When we decide to stop using palette-based coloring then it's fine to remove palette-related code, but not before then.

**Comment by jamesobutler** [comment]
The dark palette specifies for text to be white and the light palette specifies for text to be black which is what `main` is currently doing. Previously it was doing it based on the default light style palette which resulted in the text always being dark.

In this case it makes sense to define the text color to be something static and not changed dynamically. A Slicer custom app could define a Base (background) that is Blue and for Text to be green, but we would still want to define text on the slice controller differently because the background color of the slice controller is not changing with style updates. You could consider the Slicer controller decision to use black icons instead of white icons for the toolbuttons to be under the same concept. We’re not going to dynamically change those icon colors with style changes.

**Comment by lassoan** [comment]
Thanks for the explanation, it clarifies some things, but I'm still not sure where the special behavior of the spinbox text color is coming from. Why do we need to hardcode the black color of the spinbox text but not the ViewLabel ("R")? Could we color the spinbox text the same way as it is done for the ViewLabel?

**Comment by jamesobutler** [comment]
> Why do we need to hardcode the black color of the spinbox text but not the ViewLabel ("R")? Could we color the spinbox text the same way as it is done for the ViewLabel?

@lassoan Reviewing the code below shows that the Slice view label always has black text because it uses its own custom palette for the widget and does not use the application wide color palette. I could style the Spinbox with a custom palette or a custom stylesheet.

https://github.com/Slicer/Slicer/blob/31ba8942a031b907d4be2f2f5d81db717b347a1b/Libs/MRML/Widgets/qMRMLViewControllerBar.cxx#L195-L200

File: Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx [hunk 1]
```diff
@@ -409,6 +408,7 @@ void qMRMLSliceControllerWidgetPrivate::init()
     ctkDoubleSpinBox::DecimalsByShortcuts |
     ctkDoubleSpinBox::DecimalsByKey |
     ctkDoubleSpinBox::DecimalsAsMin );
+  this->SliceOffsetSlider->spinBox()->setStyleSheet("color: black; background-color: transparent;");  // Slicer light/dark mode does not manipulate Slice controller colors
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx:411]:
```suggestion
  // Slice controller background color is independent from the color palette, therefore the color of text and controls are hardcoded to black
  this->SliceOffsetSlider->spinBox()->setStyleSheet("color: black; background-color: transparent;");
```

File: Libs/MRML/Widgets/qMRMLViewControllerBar.cxx [hunk 2]
```diff
@@ -115,6 +115,7 @@ void qMRMLViewControllerBarPrivate::init()
   this->ViewLabel = new QLabel(q);
   this->ViewLabel->setObjectName("ViewLabel");
   this->ViewLabel->setAlignment(Qt::AlignHCenter | Qt::AlignCenter);
+  this->ViewLabel->setStyleSheet("color: black; background-color: transparent;");  // Slicer light/dark mode does not manipulate Slice controller colors
```
Comment (lassoan) [inline Libs/MRML/Widgets/qMRMLViewControllerBar.cxx:118]:
```suggestion
  // Slice controller background color is independent from the color palette, therefore the color of text and controls are hardcoded to black
  this->ViewLabel->setStyleSheet("color: black; background-color: transparent;");
```

**Review by lassoan — COMMENTED** [review]
OK, the implementation is now consistent and the hardcoded color is explained. Please have a look at the suggested comment changes and if it looks good then we can merge. Thank you!

File: Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx (hunk 1 above)
Comment (jamesobutler) [reply Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx]: Thanks for the updated comment suggestion. I have proceeded with this version!

**Comment by jamesobutler** [comment]
Unclear why the build step is failing do to lost connection to server?
image

**Review by lassoan — APPROVED** [review]
Thank you, it all looks good.

**Comment by lassoan** [comment]
I'll override the CI build requirement because CI does not seem to work now.

**Pull request merged**